templates/experiments/experiment_form.html

```diff
@@ -16,35 +19,62 @@
   </div>
 {% endblock %}
 {% block form %}
-  {{ form.non_field_errors }}
-  {% flag "assistants" %}
-    {% render_form_fields form "name" "description" %}
-    <h3 class="font-semibold text-xl mt-4">Language Model Configuration</h3>
-    <div class="divider divider-neutral"></div>
-    <div class="pl-4">
-      <div role="tablist" class="tabs tabs-bordered mt-4">
-        <input type="radio" name="bot_type_tabs" role="tab" class="tab" aria-label="Use LLM Directly" {% if not object.assistant_id %}checked{% endif %}/>
-        <div role="tabpanel" class="tab-content pt-4">
-          {% render_form_fields form "llm_provider" "llm" "temperature" "prompt_text" "tools_enabled" "source_material" %}
-        </div>
+  <div>
+    {% render_form_fields form "name" "description" "consent_form" %}
+    <div class="divider"></div>
+    {% render_form_fields form "type" %}
+    {% if not form.fields.assistant.choices %}
+      <div x-show="type === 'assistant'">
+        <p class="text-error">{% blocktranslate %}
+          No assistants have been created. Please create an assistant before creating an 'assistant' experiment.
+        {% endblocktranslate %}</p>
+      </div>
+    {% endif %}
 
-        <input type="radio" name="bot_type_tabs" role="tab" class="tab" aria-label="Use OpenAI Assistant" {% if object.assistant_id %}checked{% endif %} />
-        <div role="tabpanel" class="tab-content pt-4">
-          {% render_form_fields form "assistant" %}
-        </div>
+    <div role="tablist" class="tabs tabs-bordered grid-flow-col auto-cols-max mt-4">
+      <input type="radio" name="main_tabs" role="tab" class="tab" aria-label="Language Model" x-bind:checked="type === 'llm'" x-show="type === 'llm'" {% if experiment_type != "llm" %}x-cloak{% endif %}>
+      <div role="tabpanel" class="tab-content">
+        {% render_form_fields form "llm_provider" "llm" "temperature" "prompt_text" "tools_enabled" %}
+      </div>
+      <input type="radio" name="main_tabs" role="tab" class="tab" aria-label="OpenAI Assistant" x-bind:checked="type === 'assistant'" x-show="type === 'assistant'" {% if experiment_type != "assistant" %}x-cloak{% endif %}>
+      <div role="tabpanel" class="tab-content">
+        {% render_form_fields form "assistant" %}
+      </div>
+      <input type="radio" name="main_tabs" role="tab" class="tab" aria-label="Safety">
+      <div role="tabpanel" class="tab-content">
+        {% render_form_fields form "safety_layers" "safety_violation_notification_emails" "input_formatter" "max_token_limit" %}
+      </div>
+      <input type="radio" name="main_tabs" role="tab" class="tab" aria-label="Consent">
+      <div role="tabpanel" class="tab-content">
+        {% render_form_fields form "conversational_consent_enabled" %}
```

Comment:
Would it not make more sense to show the "consent" form here instead of at the top?

Reply:
That is where I had it initially but since it's required, if you leave it out there's not indication that it is the field which is preventing you from submitting since it's hidden behind a tab.